napari/utils/theme.py

```diff
@@ -13,64 +15,95 @@
     use_gradients = False
 
 from ..utils.translations import trans
-
-
-def __getattr__(attr):
-    if attr == "palettes":
-        warnings.warn(
-            trans._(
-                "palette is deprecated and will be removed after version 0.4.6. Please use get_theme and register_theme instead",
-                deferred=True,
-            ),
-            category=FutureWarning,
-            stacklevel=2,
-        )
-        return _themes
-    raise AttributeError
-
-
-_themes = {
-    'dark': {
-        'folder': 'dark',
-        'background': 'rgb(38, 41, 48)',
-        'foreground': 'rgb(65, 72, 81)',
-        'primary': 'rgb(90, 98, 108)',
-        'secondary': 'rgb(134, 142, 147)',
-        'highlight': 'rgb(106, 115, 128)',
-        'text': 'rgb(240, 241, 242)',
-        'icon': 'rgb(209, 210, 212)',
-        'warning': 'rgb(153, 18, 31)',
-        'current': 'rgb(0, 122, 204)',
-        'syntax_style': 'native',
-        'console': 'rgb(0, 0, 0)',
-        'canvas': 'black',
-    },
-    'light': {
-        'folder': 'light',
-        'background': 'rgb(239, 235, 233)',
-        'foreground': 'rgb(214, 208, 206)',
-        'primary': 'rgb(188, 184, 181)',
-        'secondary': 'rgb(150, 146, 144)',
-        'highlight': 'rgb(163, 158, 156)',
-        'text': 'rgb(59, 58, 57)',
-        'icon': 'rgb(107, 105, 103)',
-        'warning': 'rgb(255, 18, 31)',
-        'current': 'rgb(253, 240, 148)',
-        'syntax_style': 'default',
-        'console': 'rgb(255, 255, 255)',
-        'canvas': 'white',
+from .events import EventedModel
+from .events.containers._evented_dict import EventedDict
+
+THEME_KEYS = [
+    "folder",
+    "background",
+    "foreground",
+    "primary",
+    "highlight",
+    "text",
+    "icon",
+    "warning",
+    "current",
+    "syntax_style",
+    "console",
+    "canvas",
+]
+
+
+class Theme(EventedModel):
+    """Theme model."""
+
+    folder: str
```

Comment:
I agree. The `folder` attribute could be renamed to `name`. Now would be a good time to do it since we are adding a completely new model to handle theme data.

Reply:
Let's rename to "name" in this PR. But should we then reuse this "name" inside our dict of themes, i.e. should this now be the name you use when registering and getting a theme too, or can they be different?